### ISSUE TITLE
Polymorphic support

### ISSUE THREAD

**Issue by rodolfodpk** [imrafaelmerino/json-values#195]
**Is your feature request related to a problem? Please describe.**
I'd like to have specs supporting how to express classes implementing an interface: polymorphism.

**Describe the solution you'd like**
I know I can change the JSON state using lenses. I just want to know exactly what implementation is the parameter.

**Describe alternatives you've considered**
I did this before (json to a typed class implementing an interface) using [Jackson](https://github.com/crabzilla/crabzilla/blob/main/crabzilla-json-jackson/src/test/kotlin/io/github/crabzilla/jackson/JacksonJsonObjectSerDerTest.kt) and [Kotlin Serialization ](https://github.com/crabzilla/crabzilla/blob/main/crabzilla-json-kotlinx/src/test/kotlin/io/github/crabzilla/kotlinx/KotlinxJsonObjectSerDerTest.kt) I want to try to handle functions using Json Values instead, but I need to know the "type".

**Additional context**
This [article](https://json-schema.org/blog/posts/modelling-inheritance) is a good reference.

**Comment by imrafaelmerino** [comment]
take a look at the readme section **Modeling inheritance**. This is a good approach to implement it and I gave you the generators for the same price! Thanks for opening the issue. It's a real use case and I've improved some aspects... waiting for your feedback and staring the project!

**Comment by imrafaelmerino** [comment]
Can I close the issue?. Latest version 13.2.0 is needed if you want to convert that kind of specs into Avro schemas and create Avro serializers/deserializers with the library [avro-spec](https://github.com/imrafaelmerino/avro-spec) Feel free to open new issues for other common uses cases you came across

**Comment by rodolfodpk** [comment]
Yes, it can be closed. Thanks, that is exactly what I meant. I ran you example, it worked. I just noticed some warnings like: 

```
OneOf 0 trie: The key 'keyCount' has no spec associated to it. Strict specs don't allow this. Either declare de spec lenient or add a new spec for the missing key @ position=605
OneOf 0 trie: The JSON doesn't conform the spec because the required key 'buttonCount' doesn't exist @ position=759
```

But I was trying version 13.1.0. I will update to 13.2.0. Thanks!

**Comment by imrafaelmerino** [comment]
Those info messages only appear while testing because they are embedded in `assert`. OneOf specs trie to match each spec till one is valid. In that case the first spec (zero index) is invalid against the json you are parsing. json-values doesn't print anything in the console but during testing (assert enabled)
